src/track/external-captions-handler.js

```diff
@@ -311,44 +311,76 @@ class ExternalCaptionsHandler extends FakeEventTarget {
-          this._activeTextCues.push(cues[this._externalCueIndex]);
-          this._externalCueIndex++;
-          activeCuesChanged = true;
-        }
-        if (activeCuesChanged) {
-          this.dispatchEvent(new FakeEvent(CustomEventType.TEXT_CUE_CHANGED, {cues: this._activeTextCues}));
-        }
+      const isSeeked = this._isSeeked() && this._maybeSetExternalCueIndex();
+      const activeCuesRemoved = this._maybeRemoveActiveCues();
+      const activeCuesAdded = this._maybeAddToActiveCues(track);
+
+      if (isSeeked || activeCuesAdded || activeCuesRemoved) {
+        this.dispatchEvent(new FakeEvent(CustomEventType.TEXT_CUE_CHANGED, {cues: this._activeTextCues}));
       }
       // sometimes the timeupdate event is fired before the seeked event - so we need to know the user seeked.
       this._lastTimeUpdate = currentTime;
     }
   }
 
+  /**
+   * check if there was a seek. we do that because 'timeupdate' is fired before 'seeked' event.
+   * @returns {boolean} if there was a seek before
+   * @private
+   */
+  _isSeeked(): boolean {
+    return !!this._player.currentTime && Math.abs(this._player.currentTime - this._lastTimeUpdate) > 1;
+  }
+
+  /**
+   * @returns {boolean} if a cue/cues were removed from the active text cues array
+   * @private
+   */
+  _maybeRemoveActiveCues(): boolean {
+    const currentTime = this._player.currentTime;
+    if (!currentTime){
+      return false;
+    }
+    let isRemoved = false;
+    for (let activeTextCuesIndex = 0; activeTextCuesIndex < this._activeTextCues.length; activeTextCuesIndex++) {
+      const cue = this._activeTextCues[activeTextCuesIndex];
+      if (currentTime < cue.startTime || cue.endTime < currentTime) {
+        this._activeTextCues.splice(activeTextCuesIndex, 1);
+        isRemoved = true;
+      }
+    }
+    return isRemoved;
+  }
+
+  /**
+   * @param {TextTrack} track - the selected text track
+   * @returns {boolean} - if cues were added to the active text track
+   * @private
+   */
+  _maybeAddToActiveCues(track: TextTrack): boolean {
+    const currentTime = this._player.currentTime;
+    if (!currentTime){
+      return false;
+    }
+    let isAdded = false;
+    const cues = this._textTrackModel[track.language].cues;
+    while (this._externalCueIndex < cues.length && currentTime > cues[this._externalCueIndex].startTime) {
+      this._activeTextCues.push(cues[this._externalCueIndex]);
+      this._externalCueIndex++;
+      isAdded = true;
+    }
+    return isAdded;
+  }
+
   /**
    * updating the index of the text cues to the right location after a user seeked.
-   * @returns {void}
+   * @returns {boolean} if the index was changed
    * @private
    */
-  _maybeSetExternalCueIndex(): void {
+  _maybeSetExternalCueIndex(): boolean {
     const textTrack = this._player.getTracks(TrackType.TEXT).find(track => track instanceof TextTrack && track.active && track.external);
     if (textTrack && textTrack.external) {
+      this._activeTextCues = [];
```

Comment:
This looks like **magic**:tm:.
I guess you wanted to clear on seek, so it's better to be explicit and actually register to seeked and clear
https://github.com/kaltura/playkit-js/blob/9d0f0c4bbd5da4818baa65daddc84c4ebc356605/src/player.js#L1476-L1481